iothub/service/src/DigitalTwin/DigitalTwinClient.cs

```diff
@@ -36,7 +40,40 @@ public static DigitalTwinClient CreateFromConnectionString(string connectionStri
 
         private DigitalTwinClient(Uri uri, IotServiceClientCredentials credentials, params DelegatingHandler[] handlers)
         {
-            _client = new IotHubGatewayServiceAPIs(uri, credentials, handlers);
+            // ConnectionLeaseTimeoutHandler is IDisposable, so save it as a field so it can be disposed later.
+            _connectionLeaseTimeoutHandler = new ConnectionLeaseTimeoutHandler(HttpTransportSettings.DefaultConnectionLeaseTimeoutMilliseconds);
+
+            // Assign each custom delegating handler as the inner handler of the previous delegating handler
+            // For instance:
+            // _connectionLeaseTimeoutHandler.InnerHandler = handlers[0],
+            // handlers[0].InnerHandler = handlers[1],
+            // handlers[1].InnerHandler = handlers[2],
+            // ...
+            // handlers[handlers.Length].InnerHandler = _httpClientHandler
+            DelegatingHandler chainedDelegatingHandler = _connectionLeaseTimeoutHandler;
+            foreach (DelegatingHandler delegatingHandler in handlers)
+            {
+                chainedDelegatingHandler.InnerHandler = delegatingHandler;
+                chainedDelegatingHandler = delegatingHandler;
+            }
+
+            _httpClientHandler = new HttpClientHandler()
+            {
+#if !NET451
+                SslProtocols = TlsVersions.Instance.Preferred,
+                CheckCertificateRevocationList = TlsVersions.Instance.CertificateRevocationCheck,
+#endif
+            };
+
+            // The final chain in the delegating handler chain is the httpClientHandler since it is the one that actually does the network request
+            chainedDelegatingHandler.InnerHandler = _httpClientHandler;
+
+            _httpClient = new HttpClient(_connectionLeaseTimeoutHandler)
+            {
+                BaseAddress = uri
+            };
+
+            _client = new IotHubGatewayServiceAPIs(credentials, _httpClient, false);
```

Comment:
I was thinking about this a bit more, are there any other properties that get set into the http client (the one previously created by the protocol layer)? If there is, what do you think about passing the connection lease handler as a delegating handler to the constructor directly, and let the protocol layer create the http client?

Reply:
Good point. The underlying rest layer seems to generate and cache a single http client, so there isn't really a reason to create it within our hand-written layer.